File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>

using namespace std;
priority_queue<long long,vector<long long>, greater<long long>> pq;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    for(int i = 0; i < scoville.size(); i++){
        pq.push(scoville[i]);
    }
    
    while(1){
        if(pq.top() >= K || pq.size() <= 1)
            break;
        int first = pq.top();
        pq.pop();
        int second = pq.top();
        pq.pop();
        int sum = first + (second * 2);
        pq.push(sum);
        
        answer++;
    }
    if(pq.top() < K)
        return -1;
    return answer;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    static PriorityQueue<Long> pq = new PriorityQueue<>();
    
    public int solution(int[] scoville, int K) {
        int answer = 0;
        for(int i = 0; i < scoville.length; i++){
            pq.offer(Long.valueOf(scoville[i]));
        }
        
        while(!pq.isEmpty()){
            if(pq.peek() >= K){
                return answer;
            }
            Long one = pq.poll();
            if(pq.isEmpty())
                return -1;
            Long two = pq.poll();
            Long sum = one + (two * 2);
            pq.offer(sum);
            answer++;
        }
        return -1;
    }
}